.github/workflows/release_binaries.yaml

```diff
@@ -1,49 +1,67 @@
+          login-server: docker.pkg.github.com
+          username: docker
+          password: ${{ secrets.CODESIGN_GITHUB_TOKEN }}
+      - id: codesign
+        name: Pull hc-codesign image
         run: |
-          go get -u github.com/mitchellh/gox
-      - name: Checkout code
-        uses: actions/checkout@v2
-      - name: Install hc-codesign
+          docker pull docker.pkg.github.com/hashicorp/hc-codesign/hc-codesign:$VERSION
+          echo "::set-output name=image::docker.pkg.github.com/hashicorp/hc-codesign/hc-codesign:$VERSION"
+        env:
+          VERSION: v0
+      - name: Install hc-releases
         run: |
           DOWNLOAD_URL=$(curl -SsL \
             -H "Authorization: token ${{ secrets.CODESIGN_GITHUB_TOKEN }}" \
             -H "Accept: application/vnd.github.v3+json" \
-            https://api.github.com/repos/hashicorp/hc-codesign/releases/latest \
-            | jq -r '.assets[] | select(.name | contains("linux_amd64.zip")) | .url')
-          curl -SsL -o /tmp/hc-codesign.zip \
+            https://api.github.com/repos/hashicorp/hc-releases/releases/latest \
+            | jq -r '.assets[] | select(.name | contains("linux_amd64.tgz")) | .url')
+          curl -SsL -o /tmp/hc-releases.tgz \
             -H "Authorization: token ${{ secrets.CODESIGN_GITHUB_TOKEN }}" \
             -H "Accept: application/octet-stream" ${DOWNLOAD_URL}
-          unzip /tmp/hc-codesign.zip
-          sudo mv hc-codesign /usr/bin/hc-codesign
-      - name: Build binaries
+          tar xzf /tmp/hc-releases.tgz
+          sudo mv hc-releases /usr/bin/hc-releases
+      - name: Import PGP key for archive signing
+        run: echo -e "${{ secrets.PGP_SIGNING_KEY }}" | gpg --import
+      - name: Release
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --rm-dist
         env:
-          CIRCLE_TOKEN: ${{ secrets.CIRCLE_TOKEN }}
-          ARTIFACTORY_USER: ${{ secrets.ARTIFACTORY_USER }}
+          PGP_USER_ID: ${{ secrets.PGP_USER_ID }}
+          RELEASE_AWS_ACCESS_KEY_ID: ${{ secrets.RELEASE_AWS_ACCESS_KEY_ID }}
+          RELEASE_AWS_SECRET_ACCESS_KEY: ${{ secrets.RELEASE_AWS_SECRET_ACCESS_KEY }}
+          CODESIGN_IMAGE: ${{ steps.codesign.outputs.image }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           ARTIFACTORY_TOKEN: ${{ secrets.ARTIFACTORY_TOKEN }}
-        run: |
-          export PATH=$PATH:$(go env GOPATH)/bin
-          bash scripts/build_release_binaries.sh ${GITHUB_REF/refs\/tags\//}
-      - name: Release binaries
-        run: |
-          set -x
-          assets=()
-          for asset in ./release-bin/*.zip; do
-            assets+=("-a" "$asset")
-          done
-          hub release create "${assets[@]}" -m "${GITHUB_REF/refs\/tags\//}" "${GITHUB_REF/refs\/tags\//}"
+          ARTIFACTORY_USER: ${{ secrets.ARTIFACTORY_USER }}
+          CIRCLE_TOKEN: ${{ secrets.CIRCLE_TOKEN }}
+      - name: Publish released artifacts
```

Comment:
We don't want to use this yet because we're still in alpha, so we don't want to upload anything to releases.hashicorp.com. We'll want to replace this step with one that just creates a github release, like I did here: https://github.com/hashicorp/terraform-provider-kubernetes-alpha/blob/master/.github/workflows/release_binaries.yaml#L47

Reply:
edit: looks like we can do this in goreleaser itself, see here: https://goreleaser.com/customization/release/

Reply:
That's really helpful, thanks!